src/systems/user_commands/UserCommands.cc

```diff
@@ -1748,7 +1793,19 @@ bool VisualCommand::Execute()
     // When size > 1, we don't know which entity to modify
     if (entities.size() == 1)
     {
-      visualEntity = entities[0];
+      auto subentities =
+      this->iface->ecm->ChildrenByComponents(entities[0],
+        components::Name(visualMsg->name()));
+      if (subentities.size() == 1)
+      {
+        visualEntity = subentities[0];
+        gzmsg << "Using visual entity id: " << visualEntity << std::endl;
+      }
+      else
+      {
+        visualEntity = entities[0];
+        gzmsg << "Using visual entity id: " << visualEntity << std::endl;
+      }
```

Comment:
```suggestion
      auto subentities =
      this->iface->ecm->ChildrenByComponents(entities[0],
        components::Name(visualMsg->name()));
      if (subentities.size() == 1)
      {
        visualEntity = subentities[0];
      }
      else
      {
        visualEntity = entities[0];
      }
      gzmsg << "Using visual entity id: " << visualEntity << std::endl;
```

Reply:
Fair, I was going to actually give unique GZ messages to them so you could differentiate if it had another subentity. But got lazy and forgot to, probably good enough.